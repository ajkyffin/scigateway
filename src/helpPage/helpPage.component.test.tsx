import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import {
  HelpPageWithoutStyles,
  CombinedHelpPageProps,
} from './helpPage.component';

describe('Help page component', () => {
  let shallow;
  let props: CombinedHelpPageProps;

  beforeEach(() => {
    shallow = createShallow({ untilSelector: 'HelpPage' });

    props = {
      res: undefined,
      classes: {
        root: 'root-class',
        container: 'container-class',
        titleText: 'titleText-class',
        description: 'description-class',
      },
    };
  });

  it('should render correctly', () => {
    const wrapper = shallow(<HelpPageWithoutStyles {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
