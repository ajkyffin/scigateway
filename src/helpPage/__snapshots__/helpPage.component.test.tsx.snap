// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Help page component should render correctly 1`] = `
<div
  className="root-class"
>
  <WithStyles(ForwardRef(Typography))
    className="titleText-class"
    variant="h3"
  >
    title
  </WithStyles(ForwardRef(Typography))>
  <div
    className="container-class"
  >
    <WithStyles(ForwardRef(Typography))
      variant="h4"
    >
      logging-in-title
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      className="description-class"
      dangerouslySetInnerHTML={
        Object {
          "__html": "logging-in-description",
        }
      }
      variant="body1"
    />
  </div>
  <div
    className="container-class"
  >
    <WithStyles(ForwardRef(Typography))
      variant="h4"
    >
      my-data-title
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      className="description-class"
      dangerouslySetInnerHTML={
        Object {
          "__html": "my-data-description",
        }
      }
      variant="body1"
    />
  </div>
  <div
    className="container-class"
  >
    <WithStyles(ForwardRef(Typography))
      variant="h4"
    >
      browse-title
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      className="description-class"
      dangerouslySetInnerHTML={
        Object {
          "__html": "browse-description",
        }
      }
      variant="body1"
    />
  </div>
  <div
    className="container-class"
  >
    <WithStyles(ForwardRef(Typography))
      variant="h4"
    >
      search-title
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      className="description-class"
      dangerouslySetInnerHTML={
        Object {
          "__html": "search-description",
        }
      }
      variant="body1"
    />
  </div>
  <div
    className="container-class"
  >
    <WithStyles(ForwardRef(Typography))
      variant="h4"
    >
      cart-title
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      className="description-class"
      dangerouslySetInnerHTML={
        Object {
          "__html": "cart-description",
        }
      }
      variant="body1"
    />
  </div>
  <div
    className="container-class"
  >
    <WithStyles(ForwardRef(Typography))
      variant="h4"
    >
      download-title
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      className="description-class"
      dangerouslySetInnerHTML={
        Object {
          "__html": "download-description",
        }
      }
      variant="body1"
    />
  </div>
</div>
`;
